src/main/java/com/fasterxml/jackson/databind/ext/DOMSerializer.java

```diff
@@ -15,29 +16,41 @@
 import com.fasterxml.jackson.databind.jsonFormatVisitors.JsonFormatVisitorWrapper;
 import com.fasterxml.jackson.databind.ser.std.StdSerializer;
 
+import javax.xml.transform.*;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
 @SuppressWarnings("serial")
 public class DOMSerializer extends StdSerializer<Node>
 {
-    protected final DOMImplementationLS _domImpl;
+
+    private final TransformerFactory transformerFactory;
 
     public DOMSerializer() {
         super(Node.class);
-        DOMImplementationRegistry registry;
         try {
-            registry = DOMImplementationRegistry.newInstance();
+            transformerFactory = TransformerFactory.newInstance();
         } catch (Exception e) {
-            throw new IllegalStateException("Could not instantiate DOMImplementationRegistry: "+e.getMessage(), e);
+            throw new IllegalStateException("Could not instantiate TransformerFactory: "+e.getMessage(), e);
         }
-        _domImpl = (DOMImplementationLS)registry.getDOMImplementation("LS");
     }
     
     @Override
     public void serialize(Node value, JsonGenerator jgen, SerializerProvider provider)
         throws IOException
     {
-        if (_domImpl == null) throw new IllegalStateException("Could not find DOM LS");    	
-        LSSerializer writer = _domImpl.createLSSerializer();
-        jgen.writeString(writer.writeToString(value));
+        try {
+            Transformer transformer = transformerFactory.newTransformer();
+            transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+            transformer.setOutputProperty(OutputKeys.INDENT, "no");
+            StreamResult result = new StreamResult(new StringWriter());
+            DOMSource source = new DOMSource(value);
+            transformer.transform(source, result);
+            jgen.writeString(result.getWriter().toString());
+        } catch (TransformerConfigurationException e) {
+            throw new IllegalStateException("Could not create XML Transformer: "+e.getMessage(), e);
+        } catch (TransformerException e) {
+            throw new IOException("XML Transformation failed: "+e.getMessage(), e);
```

Comment:
While technically true (`IOException` can be thrown), should really throw one of `JsonProcessingException` subtypes -- see `SerializerProvider` "reportXxx()" methods (or related).